academy/4-my-own-chain/events.md

```diff
@@ -103,6 +103,165 @@ ctx.EventManager().EmitEvent(
 )
 ```
 
+## Interact with the CLI
+
+Bob made a move. Will Alice's move emit an event?
+
+```sh
+$ checkersd tx checkers play-move 0 0 5 1 4 --from $alice
+```
+
+The log is longer and admittedly, a little less readable:
+
+```
+...
+raw_log: '[{"events":[{"type":"message","attributes":[{"key":"action","value":"PlayMove"},{"key":"module","value":"checkers"},{"key":"action","value":"MovePlayed"},{"key":"Creator","value":"cosmos1gml05nvlhr0k27unas8mj827z6m77lhfpzzr3l"},{"key":"IdValue","value":"0"},{"key":"CapturedX","value":"-1"},{"key":"CapturedY","value":"-1"},{"key":"Winner","value":"NO_PLAYER"}]}]}]'
+```
+
+The expected elements are present. To parse the events and display them in a more user-friendly way, you can take the `txhash` again and do:
+
+```sh
+$ checkersd query tx 531E5708A1EFBE08D14ABF947FBC888BFC69CD6F04A589D478204BF3BA891AB7 --output json | jq ".raw_log | fromjson"
+```
+
+This returns something like:
+
+```json
+[
+  {
+    "events": [
+      {
+        "type": "message",
+        "attributes": [
+          {
+            "key": "action",
+            "value": "PlayMove"
+          },
+          {
+            "key": "module",
+            "value": "checkers"
+          },
+          {
+            "key": "action",
+            "value": "MovePlayed"
+          },
+          {
+            "key": "Creator",
+            "value": "cosmos1r80ns8496ehe73dd70r3rnr07tk23mhu2wmw66"
+          },
+          {
+            "key": "IdValue",
+            "value": "0"
+          },
+          {
+            "key": "CapturedX",
+            "value": "-1"
+          },
+          {
+            "key": "CapturedY",
+            "value": "-1"
+          },
+          {
+            "key": "Winner",
+            "value": "NO_PLAYER"
+          }
+        ]
+      }
+    ]
+  }
+]
+```
+
+As you can see, no pieces were captured. However, it turns out that Alice placed her piece ready to be captured by Bob:
+
+```sh
+$ checkersd query checkers show-stored-game 0 --output json | jq ".StoredGame.game" | sed 's/"//g' | sed 's/|/\'$'\n/g'
+*b*b*b*b
+b*b*b*b*
+***b*b*b
+**b*****
+*r******
+**r*r*r*
+*r*r*r*r
+r*r*r*r*
+```
+
+The rules of the game included in this project mandate that the player captures a piece when possible. So Bob goes ahead and captures the piece:
+
+```sh
+$ checkersd tx checkers play-move 0 2 3 0 5 --from $bob
+```
+
+It duly informs us that:
+
+```
+...
+raw_log: '[{"events":[{"type":"message","attributes":[{"key":"action","value":"PlayMove"},{"key":"module","value":"checkers"},{"key":"action","value":"MovePlayed"},{"key":"Creator","value":"cosmos1w0uumlj04eyvevhfawasm2dtjc24nexxygr8qx"},{"key":"IdValue","value":"0"},{"key":"CapturedX","value":"1"},{"key":"CapturedY","value":"4"},{"key":"Winner","value":"NO_PLAYER"}]}]}]'
+```
+
+Which formatted is:
```

Comment:
It used to be "Which, formatted, is". Now I am not sure people will understand without pausing. The below is the same as the above, except that below it has nicely formatted.

"Which formatted" would imply that the above did some formatting action, then the "is" appears and we don't understand.

Reply:
Suggested alternative: **When formatted for clarity, you see the following:**

Reply:
Yes, that's better.